src/catsDb.js

```diff
@@ -1,38 +1,72 @@
 
-const cats = [
-    {
-        id: 1,
-        name: "Catty",
-        sex: "F",
-        coat: "medium hair",
-        description: "buff or tan, skinny, talkative, often found in tall grasses or deep in the bush",
-        breedId: 1
-    },
-    {
-        id: 2,
-        name: "Frank",
-        sex: "M",
-        coat: "medium hair",
-        description: "orange, heavy-set, non-responsive except to neck scratches",
-        breedId: 1
-    },
-    {
-        id: 3,
-        name: "Pancake",
-        sex: "M",
-        coat: "shorthair",
-        description: "brown, heavy-set, spent life behind bars",
-        breedId: 1
-    },
-    {
-        id: 4,
-        name: "Madame Floof",
-        sex: "F",
-        coat: "long hair",
-        description: "white dark patches on legs, kerbside, fluffy dustbuster tail",
-        breedId: 2
-    },
-];
-
-
-module.exports = cats;
+// class 'CatRepository' serves as a template for creating new cat objects with props & methods
+
+class CatRepository {
+    constructor() {
+        this.cats = [ // array of objects - if object, change to >> this.cats:{ 1:{}, 2:{}, 3:{}}
+            {
+                id: 1,
+                name: "Catty",
+                sex: "F",
+                coat: "medium hair",
+                description: "buff or tan, skinny, talkative, often found in tall grasses or deep in the bush",
+                breedId: 1
+            },
+            {
+                id: 2,
+                name: "Frank",
+                sex: "M",
+                coat: "medium hair",
+                description: "orange, heavy-set, non-responsive except to neck scratches",
+                breedId: 1
+            },
+            {
+                id: 3,
+                name: "Pancake",
+                sex: "M",
+                coat: "shorthair",
+                description: "brown, heavy-set, spent life behind bars",
+                breedId: 1
+            },
+            {
+                id: 4,
+                name: "Madame Floof",
+                sex: "F",
+                coat: "long hair",
+                description: "white dark patches on legs, kerbside, fluffy dustbuster tail",
+                breedId: 2
+            },
+        ]
+    }
+    get getAllCats() {
+        return this.cats // returns [{},{},{}]
+        }
+
+    getCatByIndex (index) {
```

Comment:
nit: this could be named `getCatById`. Calling it an index could cause confusion as other programmers would assume that the `CatRepository` is always backed by an array - but when we update it to use an actual database it would use an ID.

Reply:
Does that mean the current 'verifyId' function that I have (takes a given 'id' to find the 'index' - to allow me to navigate the current array via index) should be a method within the Class. And correspondingly, any future database would also have its own 'convert id to index or otherwise' method?

Reply:
And nothing should refer to the index within App.js. All references should be via id. Yes?

Reply:
The implementation is fine enough as it is - it's just an issue of semantics. "Index" might be confusing in future as it has a specific meaning related to array access, whereas ID does not

Reply:
But isn't it confusing if the method is getting the cats by index and we name it getById?

Reply:
Yep - it's more an issue over naming though than anything wrong with the function. Although we discussed the other day that `verifyId` should probably move into the `CatRepository` due to [separation of concerns](https://softwareengineering.stackexchange.com/a/32614). But I'm happy for that to be done later on in another PR

Reply:
Shouldn't it really be in a separate module - and be imported into app.js? Should we be wary of dumping absolutely everything into CatRepository?

Reply:
Happy for you to organise the code how you see fit for now - we can move it around later if needed.